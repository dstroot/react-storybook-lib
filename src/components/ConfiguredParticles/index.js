import React from "react";
import Particles from "react-particles-js";
import ErrorBoundary from "../../utils/ErrorBoundary";

// styles
import "./styles.scss";

// particles configuration
// https://vincentgarreau.com/particles.js/
import config from "./particles-config.json";

const ConfiguredParticles = () => {
  return (
    <ErrorBoundary>
      <Particles params={config} className={"canvas-container"} />
    </ErrorBoundary>
  );
};

export default ConfiguredParticles;
