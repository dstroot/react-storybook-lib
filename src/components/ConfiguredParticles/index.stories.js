import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import ConfiguredParticles from '../ConfiguredParticles';

const style = {
  width: '500px',
  height: '500px',
};

// ConfiguredParticles section
storiesOf('VDS|ConfiguredParticles', module)
  .addParameters({
    info: {
      inline: true,
      header: false,
    },
  })
  .add('show ConfiguredParticles', () => (
    <div style={style}>
      <ConfiguredParticles />
    </div>
  ));
