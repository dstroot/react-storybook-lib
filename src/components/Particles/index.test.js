import React from 'react';
import renderer from 'react-test-renderer';
import ConfiguredParticles from '../Particles';

describe('ConfiguredParticles', () => {
  it('it should render', () => {
    const component = renderer.create(<ConfiguredParticles />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
