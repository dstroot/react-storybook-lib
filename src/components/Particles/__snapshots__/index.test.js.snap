// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfiguredParticles it should render 1`] = `
<div
  className="canvas-container"
>
  <canvas
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</div>
`;
